src/BaseRouter.tsx

```diff
@@ -0,0 +1,50 @@
+import { CommonAction, Router } from './types';
+
+const BaseRouter: Omit<
+  Omit<Omit<Router<CommonAction>, 'getInitialState'>, 'getRehydratedState'>,
+  'getRehydratedState'
```

Comment:
omitting twice

I think we can remove the annotation. it'll be simpler. If we annotate the individual methods, the type of the object should be inferred.

Reply:
but it's really simple, isn't it?